[mpir] build failure for x64-windows-static

### Operating system

Windows

### Compiler

MSVC

### Steps to reproduce the behavior

```Shell
vcpkg.exe install mpir:x64-windows-static
```


### Failure logs

Installing 6/7 vcpkg-msbuild:x64-windows...
Building vcpkg-msbuild:x64-windows...
-- Installing: D:/mpir/packages/vcpkg-msbuild_x64-windows/share/vcpkg-msbuild/vcpkg_msbuild.props.in
-- Installing: D:/mpir/packages/vcpkg-msbuild_x64-windows/share/vcpkg-msbuild/vcpkg_msbuild.targets.in
-- Installing: D:/mpir/packages/vcpkg-msbuild_x64-windows/share/vcpkg-msbuild/z_vcpkg_msbuild_create_props.cmake
-- Installing: D:/mpir/packages/vcpkg-msbuild_x64-windows/share/vcpkg-msbuild/vcpkg_msbuild_install.cmake
-- Installing: D:/mpir/packages/vcpkg-msbuild_x64-windows/share/vcpkg-msbuild/vcpkg-port-config.cmake
-- Installing: D:/mpir/packages/vcpkg-msbuild_x64-windows/share/vcpkg-msbuild/copyright
-- Performing post-build validation
Stored binaries in 1 destinations in 49.1 ms.
Elapsed time to handle vcpkg-msbuild:x64-windows: 124 ms
Installing 7/7 mpir:x64-windows-static...
Building mpir:x64-windows-static...
-- Downloading https://github.com/wbhart/mpir/archive/cdd444aedfcbb190f00328526ef278428702d56e.tar.gz -> wbhart-mpir-cdd444aedfcbb190f00328526ef278428702d56e.tar.gz...
-- Extracting source D:/mpir/downloads/wbhart-mpir-cdd444aedfcbb190f00328526ef278428702d56e.tar.gz
-- Applying patch enable-runtimelibrary-toggle.patch
-- Using source at D:/mpir/buildtrees/mpir/src/428702d56e-113e3f9c35.clean
-- Found external ninja('1.11.0').
-- Getting CMake variables for x64-windows-static
-- Building build.vc15/lib_mpir_gc/lib_mpir_gc.vcxproj for Release
-- Building build.vc15/lib_mpir_gc/lib_mpir_gc.vcxproj for Debug
-- Building build.vc15/lib_mpir_cxx/lib_mpir_cxx.vcxproj for Release
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: msbuild D:/mpir/buildtrees/mpir/x64-windows-static-rel/build.vc15/lib_mpir_cxx/lib_mpir_cxx.vcxproj /p:Configuration=Release /p:ForceImportAfterCppProps=D:/mpir/buildtrees/mpir/x64-windows-static/vcpkg_msbuild.props /p:ForceImportAfterCppTargets=D:/mpir/buildtrees/mpir/x64-windows-static/vcpkg_msbuild.targets /t:Rebuild /p:UseMultiToolTask=true /p:MultiProcMaxCount=21 /p:EnforceProcessCountAcrossBuilds=true /m:21 -maxCpuCount:21 /p:Platform=x64 /p:PlatformTarget=x64 /p:PlatformToolset=v143 /p:WindowsTargetPlatformVersion=10.0.22621.0 /p:VcpkgApplocalDeps=false /p:VcpkgManifestInstall=false /p:VcpkgManifestEnabled=false /p:VcpkgEnabled=false /p:VcpkgTriplet=x64-windows-static /p:VcpkgInstalledDir=D:/mpir/installed /p:WholeProgramOptimization=false
    Working Directory: D:/mpir/buildtrees/mpir/x64-windows-static-rel
    Error code: 1
    See logs for more information:
      D:\mpir\buildtrees\mpir\build-x64-windows-static-rel-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-msbuild/vcpkg_msbuild_install.cmake:108 (vcpkg_execute_required_process)
  ports/mpir/portfile.cmake:56 (vcpkg_msbuild_install)
  scripts/ports.cmake:147 (include)


error: building mpir:x64-windows-static failed with: BUILD_FAILED
Elapsed time to handle mpir:x64-windows-static: 1.9 min
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mpir
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?title=[mpir]+Build+error&body=Copy+issue+body+from+D%3A%2Fmpir%2Finstalled%2Fvcpkg%2Fissue_body.md





-------------------------


MSBuild version 17.7.2+d6990bcfa for .NET Framework
Build started 30/08/2023 15:37:32.

     1>Project "D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj" on node 1 (Rebuild target(s)).
     1>CoreClean:
         Creating directory "x64\Release\".
       PrepareForBuild:
         Creating directory "x64\Release\lib_mpir_cxx.tlog\".
       InitializeBuildStatus:
         Creating "x64\Release\lib_mpir_cxx.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
         Touching "x64\Release\lib_mpir_cxx.tlog\unsuccessfulbuild".
       MakeDirsForCl:
         Creating directory "D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\dum\my\..\..\cxx".
       ClCompile:
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64\CL.exe /c /I..\..\ /I"D:/mpir/packages/mpir_x64-windows-static/include" /I"D:/mpir/installed/x64-windows-static/include" /Zi /nologo /W1 /WX- /diagnostics:column /FS /O2 /D NDEBUG /D WIN32 /D _LIB /D HAVE_CONFIG_H /D _WIN64 /Gm- /EHsc /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Release\dum\my\..\..\cxx\\" /Fd"D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\mpirxx.pdb" /external:W1 /Gd /TP /FC /errorReport:queue "..\..\cxx\isfuns.cc"
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64\CL.exe /c /I..\..\ /I"D:/mpir/packages/mpir_x64-windows-static/include" /I"D:/mpir/installed/x64-windows-static/include" /Zi /nologo /W1 /WX- /diagnostics:column /FS /O2 /D NDEBUG /D WIN32 /D _LIB /D HAVE_CONFIG_H /D _WIN64 /Gm- /EHsc /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Release\dum\my\..\..\cxx\\" /Fd"D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\mpirxx.pdb" /external:W1 /Gd /TP /FC /errorReport:queue "..\..\cxx\ismpf.cc"
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64\CL.exe /c /I..\..\ /I"D:/mpir/packages/mpir_x64-windows-static/include" /I"D:/mpir/installed/x64-windows-static/include" /Zi /nologo /W1 /WX- /diagnostics:column /FS /O2 /D NDEBUG /D WIN32 /D _LIB /D HAVE_CONFIG_H /D _WIN64 /Gm- /EHsc /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Release\dum\my\..\..\cxx\\" /Fd"D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\mpirxx.pdb" /external:W1 /Gd /TP /FC /errorReport:queue "..\..\cxx\ismpq.cc"
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64\CL.exe /c /I..\..\ /I"D:/mpir/packages/mpir_x64-windows-static/include" /I"D:/mpir/installed/x64-windows-static/include" /Zi /nologo /W1 /WX- /diagnostics:column /FS /O2 /D NDEBUG /D WIN32 /D _LIB /D HAVE_CONFIG_H /D _WIN64 /Gm- /EHsc /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Release\dum\my\..\..\cxx\\" /Fd"D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\mpirxx.pdb" /external:W1 /Gd /TP /FC /errorReport:queue "..\..\cxx\ismpz.cc"
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64\CL.exe /c /I..\..\ /I"D:/mpir/packages/mpir_x64-windows-static/include" /I"D:/mpir/installed/x64-windows-static/include" /Zi /nologo /W1 /WX- /diagnostics:column /FS /O2 /D NDEBUG /D WIN32 /D _LIB /D HAVE_CONFIG_H /D _WIN64 /Gm- /EHsc /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Release\dum\my\..\..\cxx\\" /Fd"D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\mpirxx.pdb" /external:W1 /Gd /TP /FC /errorReport:queue "..\..\cxx\ismpznw.cc"
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64\CL.exe /c /I..\..\ /I"D:/mpir/packages/mpir_x64-windows-static/include" /I"D:/mpir/installed/x64-windows-static/include" /Zi /nologo /W1 /WX- /diagnostics:column /FS /O2 /D NDEBUG /D WIN32 /D _LIB /D HAVE_CONFIG_H /D _WIN64 /Gm- /EHsc /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Release\dum\my\..\..\cxx\\" /Fd"D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\mpirxx.pdb" /external:W1 /Gd /TP /FC /errorReport:queue "..\..\cxx\osdoprnti.cc"
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64\CL.exe /c /I..\..\ /I"D:/mpir/packages/mpir_x64-windows-static/include" /I"D:/mpir/installed/x64-windows-static/include" /Zi /nologo /W1 /WX- /diagnostics:column /FS /O2 /D NDEBUG /D WIN32 /D _LIB /D HAVE_CONFIG_H /D _WIN64 /Gm- /EHsc /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Release\dum\my\..\..\cxx\\" /Fd"D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\mpirxx.pdb" /external:W1 /Gd /TP /FC /errorReport:queue ..\..\cxx\dummy.cc
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64\CL.exe /c /I..\..\ /I"D:/mpir/packages/mpir_x64-windows-static/include" /I"D:/mpir/installed/x64-windows-static/include" /Zi /nologo /W1 /WX- /diagnostics:column /FS /O2 /D NDEBUG /D WIN32 /D _LIB /D HAVE_CONFIG_H /D _WIN64 /Gm- /EHsc /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Release\dum\my\..\..\cxx\\" /Fd"D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\mpirxx.pdb" /external:W1 /Gd /TP /FC /errorReport:queue "..\..\cxx\osfuns.cc"
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64\CL.exe /c /I..\..\ /I"D:/mpir/packages/mpir_x64-windows-static/include" /I"D:/mpir/installed/x64-windows-static/include" /Zi /nologo /W1 /WX- /diagnostics:column /FS /O2 /D NDEBUG /D WIN32 /D _LIB /D HAVE_CONFIG_H /D _WIN64 /Gm- /EHsc /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Release\dum\my\..\..\cxx\\" /Fd"D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\mpirxx.pdb" /external:W1 /Gd /TP /FC /errorReport:queue "..\..\cxx\osmpf.cc"
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64\CL.exe /c /I..\..\ /I"D:/mpir/packages/mpir_x64-windows-static/include" /I"D:/mpir/installed/x64-windows-static/include" /Zi /nologo /W1 /WX- /diagnostics:column /FS /O2 /D NDEBUG /D WIN32 /D _LIB /D HAVE_CONFIG_H /D _WIN64 /Gm- /EHsc /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Release\dum\my\..\..\cxx\\" /Fd"D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\mpirxx.pdb" /external:W1 /Gd /TP /FC /errorReport:queue "..\..\cxx\osmpq.cc"
         C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64\CL.exe /c /I..\..\ /I"D:/mpir/packages/mpir_x64-windows-static/include" /I"D:/mpir/installed/x64-windows-static/include" /Zi /nologo /W1 /WX- /diagnostics:column /FS /O2 /D NDEBUG /D WIN32 /D _LIB /D HAVE_CONFIG_H /D _WIN64 /Gm- /EHsc /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Release\dum\my\..\..\cxx\\" /Fd"D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\x64\Release\mpirxx.pdb" /external:W1 /Gd /TP /FC /errorReport:queue "..\..\cxx\osmpz.cc"
         ismpq.cc
         osdoprnti.cc
         osfuns.cc
         osmpz.cc
         osmpq.cc
         ismpz.cc
         ismpf.cc
         osmpf.cc
         dummy.cc
         isfuns.cc
         ismpznw.cc
     1>D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\osfuns.cc(30,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
     1>D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\osmpz.cc(26,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
     1>D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\osmpq.cc(26,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
     1>D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\isfuns.cc(25,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
     1>D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\osmpf.cc(29,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
     1>D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\ismpznw.cc(25,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
     1>D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\ismpz.cc(27,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
     1>D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\osdoprnti.cc(30,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
     1>D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\ismpf.cc(29,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
     1>D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\ismpq.cc(25,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
     1>Done Building Project "D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj" (Rebuild target(s)) -- FAILED.

Build FAILED.

       "D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj" (Rebuild target) (1) ->
       (ClCompile target) -> 
         D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\osfuns.cc(30,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
         D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\osmpz.cc(26,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
         D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\osmpq.cc(26,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
         D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\isfuns.cc(25,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
         D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\osmpf.cc(29,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
         D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\ismpznw.cc(25,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
         D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\ismpz.cc(27,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
         D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\osdoprnti.cc(30,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
         D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\ismpf.cc(29,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]
         D:\mpir\buildtrees\mpir\x64-windows-static-rel\cxx\ismpq.cc(25,10): fatal error C1083: Cannot open include file: 'mpir.h': No such file or directory [D:\mpir\buildtrees\mpir\x64-windows-static-rel\build.vc15\lib_mpir_cxx\lib_mpir_cxx.vcxproj]

    0 Warning(s)
    10 Error(s)

Time Elapsed 00:00:01.25


### Additional context

Might be related to https://github.com/microsoft/vcpkg/pull/33142 ? If its really failing im confused why the tests didnt find the issue.

--- a/ports/mpir/fix-static-include-files.patch
+++ b/ports/mpir/fix-static-include-files.patch
@@ -16,10 +16,10 @@ index de1ed08..e4ecceb 100644
  echo copying outputs from %tgt_dir% to %bin_dir%
  if not exist %bin_dir% md %bin_dir%
 diff --git a/build.vc15/lib_mpir_cxx/lib_mpir_cxx.vcxproj b/build.vc15/lib_mpir_cxx/lib_mpir_cxx.vcxproj
-index 3a23f01..499fe1f 100644
+index 3a23f01..1f44b22 100644
 --- a/build.vc15/lib_mpir_cxx/lib_mpir_cxx.vcxproj
 +++ b/build.vc15/lib_mpir_cxx/lib_mpir_cxx.vcxproj
-@@ -70,10 +70,17 @@
+@@ -70,6 +70,11 @@
      <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">mpirxx</TargetName>
    </PropertyGroup>
    <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
@@ -31,13 +31,7 @@ index 3a23f01..499fe1f 100644
      <ClCompile>
        <AdditionalIncludeDirectories>..\..\</AdditionalIncludeDirectories>
        <PreprocessorDefinitions>NDEBUG;WIN32;_LIB;HAVE_CONFIG_H;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-     </ClCompile>
-+    <Link>
-+    </Link>
- 
-     <PostBuildEvent>
-       <Command>cd ..\..\build.vc
-@@ -82,10 +89,17 @@ postbuild "$(TargetPath)" 15
+@@ -82,6 +87,11 @@ postbuild "$(TargetPath)" 15
      </PostBuildEvent>
    </ItemDefinitionGroup>
    <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
@@ -49,13 +43,7 @@ index 3a23f01..499fe1f 100644
      <ClCompile>
        <AdditionalIncludeDirectories>..\..\</AdditionalIncludeDirectories>
        <PreprocessorDefinitions>_DEBUG;WIN32;_LIB;HAVE_CONFIG_H;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-     </ClCompile>
-+    <Link>
-+    </Link>
- 
-     <PostBuildEvent>
-       <Command>cd ..\..\build.vc
-@@ -94,10 +108,17 @@ postbuild "$(TargetPath)" 15
+@@ -94,6 +104,11 @@ postbuild "$(TargetPath)" 15
      </PostBuildEvent>
    </ItemDefinitionGroup>
    <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
@@ -67,13 +55,7 @@ index 3a23f01..499fe1f 100644
      <ClCompile>
        <AdditionalIncludeDirectories>..\..\</AdditionalIncludeDirectories>
        <PreprocessorDefinitions>NDEBUG;WIN32;_LIB;HAVE_CONFIG_H;_WIN64;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-     </ClCompile>
-+    <Link>
-+    </Link>
- 
-     <PostBuildEvent>
-       <Command>cd ..\..\build.vc
-@@ -106,10 +127,17 @@ postbuild "$(TargetPath)" 15
+@@ -106,6 +121,11 @@ postbuild "$(TargetPath)" 15
      </PostBuildEvent>
    </ItemDefinitionGroup>
    <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
@@ -85,9 +67,3 @@ index 3a23f01..499fe1f 100644
      <ClCompile>
        <AdditionalIncludeDirectories>..\..\</AdditionalIncludeDirectories>
        <PreprocessorDefinitions>_DEBUG;WIN32;_LIB;HAVE_CONFIG_H;_WIN64;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-     </ClCompile>
-+    <Link>
-+    </Link>
- 
-     <PostBuildEvent>
-       <Command>cd ..\..\build.vc

--- a/ports/mpir/fix-static-include-files.patch
+++ b/ports/mpir/fix-static-include-files.patch
@@ -1,0 +1,93 @@
+diff --git a/build.vc/postbuild.bat b/build.vc/postbuild.bat
+index de1ed08..e4ecceb 100644
+--- a/build.vc/postbuild.bat
++++ b/build.vc/postbuild.bat
+@@ -37,12 +37,10 @@ set bin_dir="..\%extn%\%plat%\%conf%\"
+ set hdr_dir="..\%extn%\%plat%\%conf%\"
+ 
+ rem output parametrers for the MPIR tests
+-if /i "%filename%" EQU "mpirxx" goto skip 
+ echo (set ldir=%loc%)   > output_params.bat
+ echo (set libr=%extn%) >> output_params.bat
+ echo (set plat=%plat%) >> output_params.bat
+ echo (set conf=%conf%) >> output_params.bat
+-:skip
+ 
+ echo copying outputs from %tgt_dir% to %bin_dir%
+ if not exist %bin_dir% md %bin_dir%
+diff --git a/build.vc15/lib_mpir_cxx/lib_mpir_cxx.vcxproj b/build.vc15/lib_mpir_cxx/lib_mpir_cxx.vcxproj
+index 3a23f01..499fe1f 100644
+--- a/build.vc15/lib_mpir_cxx/lib_mpir_cxx.vcxproj
++++ b/build.vc15/lib_mpir_cxx/lib_mpir_cxx.vcxproj
+@@ -70,10 +70,17 @@
+     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">mpirxx</TargetName>
+   </PropertyGroup>
+   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
++    <PreBuildEvent>
++      <Command>cd ..\..\build.vc
++prebuild gc Win32 15
++      </Command>
++    </PreBuildEvent>
+     <ClCompile>
+       <AdditionalIncludeDirectories>..\..\</AdditionalIncludeDirectories>
+       <PreprocessorDefinitions>NDEBUG;WIN32;_LIB;HAVE_CONFIG_H;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+     </ClCompile>
++    <Link>
++    </Link>
+ 
+     <PostBuildEvent>
+       <Command>cd ..\..\build.vc
+@@ -82,10 +89,17 @@ postbuild "$(TargetPath)" 15
+     </PostBuildEvent>
+   </ItemDefinitionGroup>
+   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
++    <PreBuildEvent>
++      <Command>cd ..\..\build.vc
++prebuild gc Win32 15
++      </Command>
++    </PreBuildEvent>
+     <ClCompile>
+       <AdditionalIncludeDirectories>..\..\</AdditionalIncludeDirectories>
+       <PreprocessorDefinitions>_DEBUG;WIN32;_LIB;HAVE_CONFIG_H;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+     </ClCompile>
++    <Link>
++    </Link>
+ 
+     <PostBuildEvent>
+       <Command>cd ..\..\build.vc
+@@ -94,10 +108,17 @@ postbuild "$(TargetPath)" 15
+     </PostBuildEvent>
+   </ItemDefinitionGroup>
+   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
++    <PreBuildEvent>
++      <Command>cd ..\..\build.vc
++prebuild gc Win32 15
++      </Command>
++    </PreBuildEvent>
+     <ClCompile>
+       <AdditionalIncludeDirectories>..\..\</AdditionalIncludeDirectories>
+       <PreprocessorDefinitions>NDEBUG;WIN32;_LIB;HAVE_CONFIG_H;_WIN64;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+     </ClCompile>
++    <Link>
++    </Link>
+ 
+     <PostBuildEvent>
+       <Command>cd ..\..\build.vc
+@@ -106,10 +127,17 @@ postbuild "$(TargetPath)" 15
+     </PostBuildEvent>
+   </ItemDefinitionGroup>
+   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
++    <PreBuildEvent>
++      <Command>cd ..\..\build.vc
++prebuild gc Win32 15
++      </Command>
++    </PreBuildEvent>
+     <ClCompile>
+       <AdditionalIncludeDirectories>..\..\</AdditionalIncludeDirectories>
+       <PreprocessorDefinitions>_DEBUG;WIN32;_LIB;HAVE_CONFIG_H;_WIN64;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+     </ClCompile>
++    <Link>
++    </Link>
+ 
+     <PostBuildEvent>
+       <Command>cd ..\..\build.vc

--- a/ports/mpir/portfile.cmake
+++ b/ports/mpir/portfile.cmake
@@ -9,6 +9,7 @@ vcpkg_from_github(
     SHA512 f46e45bdba27c9f89953ba23186b694486fd3010bd370ea2de71a4649a2816e716a6520c9baa96936f1884437ef03f92b21c0b1fb5b757beba5a05fed30b2bfc
     HEAD_REF master
     PATCHES enable-runtimelibrary-toggle.patch
+            fix-static-include-files.patch
 )
 
 if(NOT VCPKG_TARGET_IS_WINDOWS)

--- a/ports/mpir/portfile.cmake
+++ b/ports/mpir/portfile.cmake
@@ -8,8 +8,9 @@ vcpkg_from_github(
     REF cdd444aedfcbb190f00328526ef278428702d56e # 3.0.0
     SHA512 f46e45bdba27c9f89953ba23186b694486fd3010bd370ea2de71a4649a2816e716a6520c9baa96936f1884437ef03f92b21c0b1fb5b757beba5a05fed30b2bfc
     HEAD_REF master
-    PATCHES enable-runtimelibrary-toggle.patch
-            fix-static-include-files.patch
+    PATCHES 
+        enable-runtimelibrary-toggle.patch
+        fix-static-include-files.patch
 )
 
 if(NOT VCPKG_TARGET_IS_WINDOWS)

--- a/ports/mpir/vcpkg.json
+++ b/ports/mpir/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "mpir",
   "version-date": "2022-03-02",
-  "port-version": 2,
+  "port-version": 3,
   "description": "Multiple Precision Integers and Rationals",
   "homepage": "https://github.com/wbhart/mpir",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5506,7 +5506,7 @@
     },
     "mpir": {
       "baseline": "2022-03-02",
-      "port-version": 2
+      "port-version": 3
     },
     "mpmcqueue": {
       "baseline": "2021-12-01",

--- a/versions/m-/mpir.json
+++ b/versions/m-/mpir.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "a81427e8201ddbceb242091faf87c7271fe928f7",
+      "git-tree": "66630c57080280e1089146536f7d5d21c7504600",
       "version-date": "2022-03-02",
       "port-version": 3
     },

--- a/versions/m-/mpir.json
+++ b/versions/m-/mpir.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "a81427e8201ddbceb242091faf87c7271fe928f7",
+      "version-date": "2022-03-02",
+      "port-version": 3
+    },
+    {
       "git-tree": "de9b5962358546e5cc0d7ac9b6dd9bfba95e1763",
       "version-date": "2022-03-02",
       "port-version": 2

